pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -494,14 +500,14 @@ func (e *TokensController) hasReferencedToken(serviceAccount *v1.ServiceAccount)
 	return false, nil
 }
 
-func (e *TokensController) secretUpdateNeeded(secret *v1.Secret) (bool, bool, bool) {
+func (e *TokensController) secretUpdateNeeded(secret *v1.Secret) (needsCA, needsNamespace, needsToken bool) {
 	caData := secret.Data[v1.ServiceAccountRootCAKey]
-	needsCA := len(e.rootCA) > 0 && bytes.Compare(caData, e.rootCA) != 0
+	needsCA = len(e.rootCA) > 0 && bytes.Compare(caData, e.rootCA) != 0
 
-	needsNamespace := len(secret.Data[v1.ServiceAccountNamespaceKey]) == 0
+	needsNamespace = len(secret.Data[v1.ServiceAccountNamespaceKey]) == 0
 
-	tokenData := secret.Data[v1.ServiceAccountTokenKey]
-	needsToken := len(tokenData) == 0
+	_, tokenOk, err := e.tokAuthN.AuthenticateToken(string(secret.Data[v1.ServiceAccountTokenKey]))
```

Comment:
If performance is an issue, would it be enough to just check that the key ID matches (and possibly the issuer)?